### ISSUE TITLE
Not able to add_action from within new control class

### ISSUE THREAD

**Issue by danyj** [elementor/elementor#8045]
I created new control but need to run some wp_ajax from it . 
I tried to add action from control `__construct` 

```
     public function __construct(){
	parent::__construct();
         add_action('wp_ajax_some_handle',  array( $this , '_ajax_method' ) ); 
     }
```
or after the class with 

`add_action( 'wp_ajax_some_handle', array( "My_Class_Name", '_ajax_method' ) )`

It does not work , 
only way to add action is within an external file that is included before 

`elementor/controls/controls_registered`

which forces me to separate control in to multiple files and classes with out any reason

The docs do not have any info about it or init/ construct file inclusions either

https://developers.elementor.com/creating-a-new-control/

basically we have no way of adding any actions from the controls.

**Comment by danielflippance** [comment]
Same issue for me when trying to add AJAX calls to a custom Elementor widget - The HTTP call responds with a 400 error. The code works when moved to functions.php but not when nicely encapsulated within the widget.

**Comment by danyj** [comment]
widgets have something, I think is  
 `register_ajax_action`  
`add_action( 'elementor/ajax/register_actions', [ $this, 'register_ajax_actions' ] );`

 look it up in the pro ,  but controls dont have it

**Comment by danyj** [comment]
@danielflippance  figured it out,

think like this , everything is module first,
so your module manager can load either widget or  control so create a module first and within register the control  and than you can add ajax actions or any actions from that module 

something like this 

```

<?php
/**
 * @package      Thz Framework
 * @author       Themezly
 * @websites     http://www.themezly.com | http://www.youjoomla.com | http://www.yjsimplegrid.com
 */

namespace ThemezlyElements\Modules\Controls;

use ThemezlyElements\Base\Module_Base;
use ThemezlyElements\Modules\Controls\MyControlNameFolder\MyControlClass;

if ( ! defined( 'ABSPATH' ) ) {
	exit; // Exit if accessed directly.
}

/**
 * Class Module.
 */
class Module extends Module_Base {
	
	const MY_CONTROL_ID = 'my-control-name';

	/**
	 * Get Module Name.
	 *
	 * @since 0.0.1
	 * @access public
	 *
	 * @return string Module name.
	 */
	public function get_name() {
		return 'custom-controls';
	}

	/**
	 * Constructor.
	 */
	public function __construct() {
		parent::__construct();
		$this->add_actions();
	}
	
	
	public function _my_ajax_one(){
		$response = array(
			'message' => '_my_ajax_one',
		);
		wp_send_json_success( $response );		
	}
	
	public function _my_ajax_two(){
		$response = array(
			'message' => '_my_ajax_two',
		);
		wp_send_json_success( $response );		
	}	
	
	protected function add_actions() {

		add_action( 'wp_ajax_my_aj_handle_one', array( $this, '_my_ajax_one' ) );
		add_action( 'wp_ajax_my_aj_handle_two', array( $this, '_render' ) );

		add_action( 'elementor/controls/controls_registered', array ( $this, '_my_ajax_two' ) );

	}
		
	public function register_controls() {
		
		$controls_manager = \Elementor\Plugin::$instance->controls_manager;

		$controls_manager->register_control( self::MY_CONTROL_ID, new MyControlClass() );
	}

}
```

**Comment by notrealdev** [comment]
Same issue for me...
